pkg/derivation/hashes.go

```diff
@@ -141,6 +141,18 @@ func (d *Derivation) CalculateOutputPaths(inputDrvReplacements map[string]string
 // We solve this having calculateDrvReplacement accept a map of
 // /its/ replacements, instead of recursing.
```

Comment:
The diff is a bit hard to read, but this docstring does explicitly say we expect a (immutabe) hashmap with the hashes already precomputed. Doing a call to `CalculateDrvReplacementRecursive` inside then seems wrong.